test/breakpoints.jl

```diff
@@ -322,3 +322,39 @@ end
     frame, bp = @interpret sin(2.0)
     @test bp isa BreakpointRef
 end
+
+const breakpoint_update_hooks = JuliaInterpreter.breakpoint_update_hooks
+@testset "hooks" begin
+    remove()
+    f_break(x) = x
+
+    # Check creating hits hook
+    empty!(breakpoint_update_hooks)
+    hook_hit = false
+    push!(breakpoint_update_hooks, (f,_)->hook_hit = f == breakpoint)
+    bp = breakpoint(f_break)
+    @test hook_hit
```

Comment:
Probably want to test one that doesn't hit first.

Reply:
I'm not sure what that would like like

Reply:
Do something to breakpoints that isn't always caught by your hook, and then `@test !hook_hit`.

Reply:
Such as?
What code am I trying to test, for what failure case?

 - Do I want to test the definition of `(f,_)->hook_hit = f == breakpoint` ? (so I should test e.g. by doing a `remove`)
 - Do I want to test that random functions don't trigger hooks, so I should test by doing `1+1` or something?